app/Http/Controllers/StudentController.php

```diff
@@ -95,28 +95,53 @@ public function deleteCourse($id)
 
     public function listCourse(Request $request)
     {
-        if ($request->has('search_course')){
+        if ($request->has('search_course')) {
             $student = Auth::guard('student')->user();
             $search = $request->search_course;
             $courses = Course::whereDoesntHave('students', function ($query) use ($student) {
-                $query->where('students.id', $student->id);})->whereHas('subject',function ($query) use ($search) {
-                $query->where('name','like','%'.$search.'%');
-            })->orWhere('course_code','like','%'.$search.'%')->get();
-            return view('system.student.list_course', ['courses' => $courses,'student'=>$student]);
-        }else {
+                $query->where('students.id', $student->id);
+            })->where('course_code', 'like', '%' . $search . '%')->orWhereHas('subject', function ($query) use ($search) {
+                $query->where('name', 'like', '%' . $search . '%');
+            })->get();
+            return view('system.student.list_course', ['courses' => $courses, 'student' => $student]);
+        } else {
             $student = Auth::guard('student')->user();
             $courses = Course::whereDoesntHave('students', function ($query) use ($student) {
                 $query->where('students.id', $student->id);
             })->with('subject')->get();
-            return view('system.student.list_course', ['courses' => $courses,'student'=>$student]);
+            return view('system.student.list_course', ['courses' => $courses, 'student' => $student]);
         }
     }
 
     public function registerCourseStudent($id)
     {
         $student = Auth::guard('student')->user();
-        $student->courses()->attach($id);
-        return redirect()->route('student');
+        $course = Course::findOrFail($id);
+        $courseSelect = $student->courses()->where('weekday', $course->weekday)->orderBy('time_start', 'asc')->get();
+        if ($student->courses()->where('course_id', $id)->exists()) {
+            return back()->withErrors('you registered this course');
+        } elseif (count($courseSelect) > 1) {
+            for ($i = 0; $i < count($courseSelect); $i++) {
```

Comment:
đoan này anh thấy thay bằng foreach hay hơn. không vần phải check count > 1 rồi for lằng nhằng

Reply:
cái này e làm nó giải quyết theo từng số lượng phần tử mảng query được nên phải xét từng trường hợp v anh ạ